src/sidebar/views/overview.ts

```diff
@@ -1,15 +1,26 @@
-import { Command, Event, EventEmitter, ExtensionContext, TreeDataProvider, TreeItem, TreeView, window } from 'vscode';
+import {
+    Command,
+    Event,
+    EventEmitter,
+    ExtensionContext,
+    ThemeIcon,
+    TreeDataProvider,
+    TreeItem,
+    TreeView,
+    window
+} from 'vscode';
 import { ExtensionApi } from '../../backend/api';
 import { CheckerMetadata } from '../../backend/types';
 
 export class OverviewItem {
-    constructor(private label: string | (() => string), private command?: Command) {}
+    constructor(private label: string | (() => string), private command?: Command, private iconPath?: ThemeIcon) {}
```

Comment:
Might want to swap the order of the args, since icons are much more common than commands now.

Another small thing would be to make the ThemeIcon inside the constructor, with a string as the iconPath, instead of taking the object as the arg.

Reply:
Yes, good idea, this way also the code becomes simple.